go-controller/pkg/clustermanager/egressip_controller_test.go

```diff
@@ -36,13 +37,18 @@ type fakeEgressIPHealthClient struct {
 	Connected        bool
 	ProbeCount       int
 	FakeProbeFailure bool
+	mutex            sync.Mutex
```

Comment:
The race fix in the UT's looks good to me, thanks for taking care of it @pperiyasamy.

Reply:
Thanks @pperiyasamy and @kyrtapz

Reply:
Done